antarest/study/storage/variantstudy/variant_study_service.py

```diff
@@ -971,20 +971,28 @@ def _safe_generation(self, metadata: VariantStudy, timeout: int = DEFAULT_AWAIT_
             task_id = self.generate_task(metadata)
             self.task_service.await_task(task_id, timeout)
             result = self.task_service.status_task(task_id, RequestParameters(DEFAULT_ADMIN_USER))
-            if result.result and result.result.success:
+            if not result.result:
+                raise ValueError("No task result")
+            if result.result.success:
                 # OK, the study has been generated
                 return
-            raise ValueError("No task result or result failed")
+            # The variant generation failed, we have to raise a clear exception.
+            error_msg = result.result.message
+            stripped_msg = error_msg.removeprefix(
+                f"Task {task_id} failed: Unhandled exception 417: Failed to generate variant study {metadata.id}"
```

Comment:
It's a little fragile to do this, if we change one letter in the base message, this will not work anymore: do you think we could do something more robust?